src/ros/controller.cpp

```diff
@@ -101,18 +101,27 @@ void ROSController::read(RTShared& packet)
   joint_interface_.update(packet);
   wrench_interface_.update(packet);
   robot_state_received_ = true;
+  most_recent_packet_time_ = ros::Time::now();
 }
 
 bool ROSController::update()
 {
   // don't run controllers if we haven't received robot state yet
   if (!robot_state_received_)
+  {
+    ROS_INFO_THROTTLE(1.0, "No messages received from the robot so far.");
     return true;
+  }
 
   auto time = ros::Time::now();
   auto diff = time - lastUpdate_;
   lastUpdate_ = time;
 
+  if ((time - most_recent_packet_time_).toSec() > 0.01)  // CB2 and 3 cycle times are 0.008, adding some margin
```

Comment:
I'd like to suggest increasing the [pipeline timeout](https://github.com/ros-industrial/ur_modern_driver/blob/048edd205af824b22410323723ac373327a8973b/include/ur_modern_driver/pipeline.h#L183) a bit and getting rid of this check by moving the warning to [`onTimeout()`](https://github.com/ros-industrial/ur_modern_driver/blob/048edd205af824b22410323723ac373327a8973b/src/ros/controller.cpp#L136). What do you think?

Reply:
That sounds like a very cool solution. Howeer, [the comment on the pipeline timeout](https://github.com/ros-industrial/ur_modern_driver/blob/048edd205af824b22410323723ac373327a8973b/include/ur_modern_driver/pipeline.h#L179) states the reasoning for this is to update the consumer with the cycle time also if no packets have been received. So increasing this timeout seems to be against this idea.

Reply:
Revisiting [previous discussions](https://github.com/ros-industrial/ur_modern_driver/issues/206) I think the only reason to force an controller update/write cycle every 8ms is [this `do_brake` logic](https://github.com/ros-industrial/ur_modern_driver/blob/048edd205af824b22410323723ac373327a8973b/src/ros/trajectory_follower.cpp#L51-L59) in the URScript program. I'm not 100% onboard with this decision and, furthermore, this hasn't been changed in the UR-e version and apparently it isn't causing any issue.

It's true, though, that updating the pipeline timeout should be accompanied by some modification to the URScript, possibly removing the `stopj` altogether (the robot should just stop at the last `servoj` commanded position, anyways). I understand you may want to avoid changing that, so feel free to explore that path or to ignore my suggestion and resolve this.

Reply:
Then I'll resolve this for now.